tests/sles4sap/redirection_tests/ensa2_kill_sapinstance.pm

```diff
@@ -87,32 +100,35 @@ sub run {
 
     # Kill sapinstance process
     my $process_name;
-    $process_name = 'en.sap' if $instance_type eq 'ASCS';
-    $process_name = 'er.sap' if $instance_type eq 'ERS';
+    # There is a different naming between S4HANA and regular NW installation.
+    # Note: 'pgrep' accepts only limited regexes
+    $process_name = '"en.sap|enq.sap"' if $instance_type eq 'ASCS';
+    $process_name = '"er.sap|enqr.sap"' if $instance_type eq 'ERS';
     die "Unknown instance type: '$instance_type'" unless $process_name;
 
-    record_info('PROC list', script_output("ps -ef | grep $process_name"));    # Show SAP processes running
-    my $pid = script_output("pgrep $process_name");
+    record_info('PROC list', script_output("ps -ef | grep -E $process_name"));    # Show SAP processes running
```

Comment:
Why not to run pgrep here too?

Reply:
Hmm... I thought pgrep only lists PIDs, but it does process name as well with `pgrep -a`. 
I'm going to change it but will do only a short VR on already existing cluster I have. 
Thanks for the suggestion.

Reply:
Ok... I was trying using pgrep but it's completely weird with using regexes. 
I'll go back to ps to avoid problems later.